components/server/ee/src/workspace/gitpod-server-impl.ts

```diff
@@ -2146,35 +2146,64 @@ export class GitpodServerEEImpl extends GitpodServerImpl {
         }
     }
 
-    async getUsageLimitForTeam(ctx: TraceContext, teamId: string): Promise<number | undefined> {
-        const user = this.checkAndBlockUser("getUsageLimitForTeam");
-        const team = await this.guardTeamOperation(teamId, "get");
-        await this.ensureStripeApiIsAllowed({ team });
+    async getUsageLimitFor(ctx: TraceContext, attributionId: string): Promise<number | undefined> {
+        const attrId = AttributionId.parse(attributionId);
+        if (attrId == undefined) {
+            log.error(`Invalid attribution id: ${attributionId}`);
+            throw new ResponseError(ErrorCodes.BAD_REQUEST, `Invalid attibution id: ${attributionId}`);
+        }
 
-        const attributionId = AttributionId.render({ kind: "team", teamId });
+        const user = this.checkAndBlockUser("getUsageLimitFor");
+        switch (attrId.kind) {
+            case "team":
+                const team = await this.guardTeamOperation(attrId.teamId, "get");
```

Comment:
This should be a `guardCostCenter`, as this is an operation on that entity.

Reply:
This is done below the `switch` on line 2166:

https://github.com/gitpod-io/gitpod/blob/3a730199be6aacbde3d999a6bba8cc50046e386f/components/server/ee/src/workspace/gitpod-server-impl.ts#L2166

Reply:
Thx, noticed [here](https://github.com/gitpod-io/gitpod/pull/12767#discussion_r966694409), forgot to update. :+1: